### [CVE-2022-29203](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-29203)
![](https://img.shields.io/static/v1?label=Product&message=tensorflow&color=blue)
![](https://img.shields.io/static/v1?label=Version&message=%3C%202.6.4%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=%3E%3D%202.7.0rc0%2C%20%3C%202.7.2%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=%3E%3D%202.8.0rc0%2C%20%3C%202.8.1%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=%3E%3D%202.9.0rc0%2C%20%3C%202.9.0%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Vulnerability&message=CWE-190%3A%20Integer%20Overflow%20or%20Wraparound&color=brightgreen)

### Description

TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.SpaceToBatchND` (in all backends such as XLA and handwritten kernels) is vulnerable to an integer overflow: The result of this integer overflow is used to allocate the output tensor, hence we get a denial of service via a `CHECK`-failure (assertion failure), as in TFSA-2021-198. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.

### POC

#### Reference
- https://github.com/tensorflow/tensorflow/releases/tag/v2.7.2

#### Github
- https://github.com/gclonly/im

